plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/TestTransactionScopeCachingDirectoryLister.java

```diff
@@ -96,7 +98,7 @@ public void testConcurrentDirectoryListing()
                         path1, ImmutableList.of(firstFile, secondFile),
                         path2, ImmutableList.of(thirdFile)));
 
-        TransactionScopeCachingDirectoryLister cachingLister = new TransactionScopeCachingDirectoryLister(countingLister, 2);
+        TransactionScopeCachingDirectoryLister cachingLister = new TransactionScopeCachingDirectoryLister(countingLister, DataSize.ofBytes(1500), Optional.of(1));
```

Comment:
It's non obvious why `testConcurrentDirectoryListing` overrides cache's concurrency level. Shouldn't it test how the default setup behaves in concurrent context?

Please explain.

Reply:
Actually, if i remove `concurrencyLevel`, the first failing assertion is
```
assertThat(cachingLister.isCached(path2)).isTrue();
```

which looks to prove that the cache doesn't work (with default concurrencyLevel)

Reply:
Per `com.google.common.cache.CacheBuilder#maximumWeight`:
```
Note that the cache may evict an entry before this limit is exceeded. For example, in the current implementation, when concurrencyLevel is greater than 1, each resulting segment inside the cache independently limits its own weight to approximately maximumWeight / concurrencyLevel.
```

cache with higher concurrency limit is not deterministic

added comment